python/cudf/cudf/tests/indexes/datetime/test_time_specific.py

```diff
@@ -1,29 +1,28 @@
 # Copyright (c) 2022-2024, NVIDIA CORPORATION.
+import zoneinfo
+
 import pandas as pd
 
 import cudf
 from cudf.testing._utils import assert_eq
 
 
 def test_tz_localize():
+    tz = zoneinfo.ZoneInfo("America/New_York")
     pidx = pd.date_range("2001-01-01", "2001-01-02", freq="1s")
     pidx = pidx.astype("<M8[ns]")
     idx = cudf.from_pandas(pidx)
     assert pidx.dtype == idx.dtype
-    assert_eq(
-        pidx.tz_localize("America/New_York"),
-        idx.tz_localize("America/New_York"),
-    )
+    assert_eq(pidx.tz_localize(tz), idx.tz_localize(tz))
```

Comment:
*question*: Does cudf support tz_localize of "ambiguous/non-existent" datetimes? If yes, should we test? If no, should we test with xfailures and an issue?

Reply:
Yeah we test this in `python/cudf/cudf/tests/series/test_datetimelike.py::test_localize_ambiguous` and `python/cudf/cudf/tests/series/test_datetimelike.py::test_localize_nonexistent` and translating those cases to `NA`